db.go

```diff
@@ -96,6 +96,12 @@ type Db struct {
 	fwdHistoryLimit int
 }
 
+func dbWithCustomForwardingHistoryLimit(limit int) func(d *Db) {
```

Comment:
Maybe this can now just be a normal parameter (not an option), with the default set at the cli level?